polyfill/lib/ecmascript.mjs

```diff
@@ -1407,6 +1410,19 @@ export const ES = ObjectAssign({}, ES2019, {
     return { years, months, weeks, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds };
   },
 
+  CreateTemporalDate: (isoYear, isoMonth, isoDay, calendar, object) => {
+    ES.RejectDate(isoYear, isoMonth, isoDay);
+    ES.RejectDateRange(isoYear, isoMonth, isoDay);
+    if (!calendar || typeof calendar !== 'object') throw new RangeError('invalid calendar');
+    if (object === undefined) object = new GetIntrinsic('%Temporal.Date%');
```

Comment:
no, this works.

Reply:
It wouldn't be an infinite loop, because the constructor never calls it with an undefined `object`. That also means this is dead code, so I don't know how you verified it works. Let's either remove the undefined case for now, or re-inline all this into the constructor.